util/actions.js

```diff
@@ -0,0 +1,27 @@
+'use server';
+import { cookies } from 'next/headers';
+import { getFruitByName } from '../database/fruits';
+import { getCookieByName } from './cookies';
+
+export async function setFruitNote(formData) {
+  const singleFruit = getFruitByName(formData.get('fruit-name'));
+
+  if (!singleFruit) return;
```

Comment:
this can be simplified by changing the `formData` to `id` and passing it similar to [this example in the docs](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#:~:text=addOptimisticMessage(message)%3B):

```js
'use client';
 
import { experimental_useOptimistic as useOptimistic } from 'react';
import { send } from './actions.js';
 
export function Thread({ messages }) {
  const [optimisticMessages, addOptimisticMessage] = useOptimistic(
    messages,
    (state, newMessage) => [...state, { message: newMessage, sending: true }],
  );
  const formRef = useRef();
 
  return (
    <div>
      {optimisticMessages.map((m) => (
        <div>
          {m.message}
          {m.sending ? 'Sending...' : ''}
        </div>
      ))}
      <form
        action={async (formData) => {
          const message = formData.get('message');
          formRef.current.reset();
          addOptimisticMessage(message);
          await send(message);
        }}
        ref={formRef}
      >
        <input type="text" name="message" />
      </form>
    </div>
  );
}
```

Reply:
https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/blob/update-fruit-example-cookies/app/fruits/%5BfruitId%5D/actions.js